File: src/main/java/com/example/healthylifestyleapp/FragmentLogin.java
package com.example.healthylifestyleapp;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatButton;
import androidx.appcompat.widget.AppCompatEditText;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.FirebaseApp;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

public class FragmentLogin extends android.app.Fragment implements View.OnClickListener {

    AppCompatEditText AppCompatEditTextUserName, AppCompatEditTextPassword;
    AppCompatButton AppCompatButtonLogIn;
    //firebase auth object
    private FirebaseAuth firebaseAuth;

    //progress dialog
   // private ProgressDialog progressDialog;
   ProgressBar progressBar;

    public View onCreateView(LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        //Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.fragment_login, container, false);

        AppCompatEditTextUserName = (AppCompatEditText) rootView.findViewById(R.id.AppCompatEditTextUserName);
        AppCompatEditTextPassword = (AppCompatEditText) rootView.findViewById(R.id.AppCompatEditTextPassword);
        AppCompatButtonLogIn = (AppCompatButton) rootView.findViewById(R.id.AppCompatButtonLogIn);

        progressBar = rootView.findViewById(R.id.progressBar);

        //attaching click listener
        AppCompatButtonLogIn.setOnClickListener(this);

//getting firebase auth object
        firebaseAuth = FirebaseAuth.getInstance();
        FirebaseApp.initializeApp(getActivity());

        //if the objects getcurrentuser method is not null
        //means user is already logged in
        /*if (firebaseAuth.getCurrentUser() != null) {
            //close this activity
            getActivity().finish();
            //opening profile activity
            //startActivity(new Intent(getActivity(), StartedActivity.class));


        }*/

        return rootView;
    }
    @Override
    public void onClick(View v) {
        switch (v.getId())
        {
            case R.id.AppCompatButtonLogIn:

                String username=AppCompatEditTextUserName.getText().toString().trim();
                String password = AppCompatEditTextPassword.getText().toString().trim();
                if (TextUtils.isEmpty(password)) {
                    Toast.makeText(getActivity(), "Enter password!", Toast.LENGTH_SHORT).show();
                    return;
                }

               /* if (password.length() < 6) {
                    Toast.makeText(getActivity(), "Password too short, enter minimum 6 characters!", Toast.LENGTH_SHORT).show();
                    return;
                }
*/
                if (TextUtils.isEmpty(username)) {
                    Toast.makeText(getActivity(), "Enter Username!", Toast.LENGTH_SHORT).show();
                    return;
                }

                progressBar.setVisibility(View.VISIBLE);
               /* firebaseAuth.signInWithEmailAndPassword(username, password)
                        .addOnCompleteListener(getActivity(), new OnCompleteListener<AuthResult>() {
                            @Override
                            public void onComplete(@NonNull Task<AuthResult> task) {
                                Toast.makeText(getActivity(),"createUserWithEmail:onComplete:" + task.isSuccessful(), Toast.LENGTH_SHORT).show();

                                progressBar.setVisibility(View.GONE);

                                if (!task.isSuccessful()) {
                                    Toast.makeText(getActivity(), "Enter Correct Username and Password" + task.getException(),
                                            Toast.LENGTH_SHORT).show();
                                } else {
                                    Toast.makeText(getActivity(), "Login Successful!", Toast.LENGTH_LONG).show();
                                    progressBar.setVisibility(View.GONE);
                                    Intent myIntent=new Intent(getActivity(), ProfileScreenActivity.class);
                                    startActivity(myIntent);

                                   *//* return inflater.inflate(
                                            R.layout.fragment_login, container, false);*//*
                                    // startActivity(new Intent(getActivity(), MainActivity.class));
                                }


                            }
                        });*/

                firebaseAuth.signInWithEmailAndPassword(username, password)
                        .addOnCompleteListener(getActivity(), new OnCompleteListener<AuthResult>() {
                            @Override
                            public void onComplete(@NonNull Task<AuthResult> task) {
                                Toast.makeText(getActivity(),"createUserWithEmail:onComplete:" + task.isSuccessful(), Toast.LENGTH_SHORT).show();

                                progressBar.setVisibility(View.GONE);
                                if (!task.isSuccessful()) {
                                    Toast.makeText(getActivity(), "Enter Correct Username and Password" + task.getException(),
                                            Toast.LENGTH_SHORT).show();
                                } else {
                                    Toast.makeText(getActivity(), "Login Successful!", Toast.LENGTH_LONG).show();
                                    progressBar.setVisibility(View.GONE);
                                    Intent myIntent=new Intent(getActivity(), StartedActivity.class);
                                    startActivity(myIntent);

                                    // startActivity(new Intent(getActivity(), MainActivity.class));
                                }


                            }
                        });

        }


    }

    @Override
    public void onResume() {
        super.onResume();
        progressBar.setVisibility(View.GONE);
    }

}

File: app/src/main/java/com/example/healthylifestyleapp/ui/fragments/FragmentLogin.kt
package com.example.healthylifestyleapp.ui.fragments

import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.crashlytics.android.Crashlytics
import com.example.healthylifestyleapp.R
import com.example.healthylifestyleapp.ui.activities.ForgotPasswordActivity
import com.example.healthylifestyleapp.ui.activities.base.activity.BaseActivity
import com.example.healthylifestyleapp.ui.activities.base.fragment.BaseFragment
import com.example.healthylifestyleapp.utils.next
import kotlinx.android.synthetic.main.fragment_login.*
import org.jetbrains.anko.support.v4.startActivity
import org.jetbrains.anko.support.v4.toast

class FragmentLogin : BaseFragment(), View.OnClickListener {
    override fun getRoot(): View? {
        return rootView
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_login, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        //attaching click listener
        AppCompatButtonLogIn.setOnClickListener(this)

        AppCompatTextViewForgotPassword.setOnClickListener {
            activity?.finish()
            startActivity<ForgotPasswordActivity>()
        }
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.AppCompatButtonLogIn -> {

                val username = AppCompatEditTextUserName.text!!.toString().trim { it <= ' ' }
                val password = AppCompatEditTextPassword.text!!.toString().trim { it <= ' ' }
                if (TextUtils.isEmpty(password)) {
                    Toast.makeText(activity, "Enter password!", Toast.LENGTH_SHORT).show()
                    return
                }
                if (TextUtils.isEmpty(username)) {
                    Toast.makeText(activity, "Enter Username!", Toast.LENGTH_SHORT).show()
                    return
                }


                showProgressDialog()
                firebaseAuth.signInWithEmailAndPassword(username, password)
                    .addOnCompleteListener(activity!!) { task ->
                        Toast.makeText(
                            activity,
                            "createUserWithEmail:onComplete:" + task.isSuccessful,
                            Toast.LENGTH_LONG
                        ).show()

                        dismissProgressDialog()
                        if (task.isSuccessful) {
                            Toast.makeText(activity, "Login Successful!", Toast.LENGTH_LONG).show()
                            progressBar.visibility = View.GONE
                            (activity as BaseActivity).next()
                        }
                    }
                    .addOnFailureListener {
                        dismissProgressDialog()
                        toast(it.message!!)
                        it.printStackTrace()
                        Crashlytics.logException(it)
                    }
            }
        }


    }

    override fun onResume() {
        super.onResume()
        progressBar.visibility = View.GONE
    }

}